src/NEPTransformations.jl

```diff
@@ -0,0 +1,294 @@
+    the use of `MobiusTransformedNEP` can
+    considerably slow down NEP-access.
+
+    # Example
+
+    ```julia-repl
+    julia> nep0=nep_gallery("pep0")
+    julia> a=1; b=3; c=4; d=5;
+    julia> nep1=mobius_transform(nep0,a=a,b=b,c=c,d=d);
+    julia> s=3;
+    julia> opnorm(compute_Mder(nep0,(a*s+b)/(c*s+d))-compute_Mder(nep1,s))
+    0.0
+    ```
+    """
+    function mobius_transform(orgnep::NEP;a=1,b=0,c=0,d=1)
+        return MobiusTransformedNEP(orgnep;a=a,b=b,c=c,d=d)
+    end
+
+    function mobius_transform(orgnep::SPMF_NEP;a=1,b=0,c=0,d=1)
+        orgfv=get_fv(orgnep);
+        m=size(orgfv,1);
+        fv = Vector{Function}(undef, m)
+        # create anonymous functions corresponding to the
+        # möbius transformed problem
+        for i=1:m
+            fv[i] = S -> orgfv[i]((a*S + b*one(S)) / (c*S + d*one(S)))::((S isa Number) ? Number : Matrix)
+        end
+        # create a new SPMF with transformed functions
+        return SPMF_NEP(get_Av(orgnep),fv;Schur_fact=orgnep.Schur_factorize_before);
+    end
+
+
+
+    """
+    Hidden type representing a transformed NEP
+    """
+    struct MobiusTransformedNEP <: NEP
+        a::Number
+        b::Number
+        c::Number
+        d::Number
+        orgnep::NEP
+        function MobiusTransformedNEP(orgnep;a=1,b=0,c=0,d=1)
+            return new(a,b,c,d,orgnep)
+        end
+    end
+
+    # Size and issparse implemented by delegation
+    function size(nep::MobiusTransformedNEP,dim=-1)
+        return size(nep.orgnep,dim)
+    end
+    function issparse(nep::MobiusTransformedNEP)
+        return issparse(nep.orgnep)
+    end
+
+    function compute_Mder(nep::MobiusTransformedNEP,λ::Number,i::Integer=0)
+        # I did not found a better way
+        return compute_Mder_from_MM(nep,λ,i);
+    end
+
+    function compute_MM(nep::MobiusTransformedNEP,S,V)
+        # Just call orgnep with a different S
+        return compute_MM(nep.orgnep, (nep.a*S + nep.b*I) / (nep.c*S + nep.d*I), V)
+    end
+
+    function compute_Mlincomb(nep::MobiusTransformedNEP,λ::Number,V::AbstractVecOrMat
+                              ,a::Vector=ones(size(V,2)))
+        # I did not found a better way
+        return compute_Mlincomb_from_MM(nep,λ,V,a)
+    end
+
+
+    # consider renaming this function
+    """
+        transform_to_pep(orgnep::NEP[,d=2])
+
+    Compute the truncated (with `d` terms) Taylor series of the NEP.
+    The output is a [`PEP`](@ref).
+    """
+    function transform_to_pep(nep::NEP,d::Integer=2)
+        A=Array{Array{Float64, 2}}(d+1)
+        for i=0:d
+            A[i+1]=compute_Mder(nep,0,i)/factorial(i);
+        end
+        return PEP(A)
+    end
+
+    # representation of the structured linearizations used in the CORK framework
+    struct CORK_pencil{T<:AbstractMatrix}
+        M::T
+        N::T
+        Av::Vector{T}   # Array of Array of matrices
+        Bv::Vector{T}   # Array of Array of matrices
+    end
```

Comment:
I think you want the matrix-types to be decoupled. If the problem is a full matrix, you may still want to represent the other matrices in sparse. 
```
struct CORK_pencil{TM<:AbstractMatrix,TN<:AbstractMatrix,TAv<:AbstractMatrix,TBv<:AbstractMatrix}
    M::TM
...
```

Reply:
We have discussed this issue a couple of times IRL and you convinced me, but I have now changed my mind during the implementation. The reason is the following:

Each matrix in the vector `Av` and `Bv` are computed as linear combination of matrices generated with `compute_Mder`. Therefore it makes sense that the type of `Av` and `Bv` is inherited by the type of the `nep`. So, if the `nep` is sparse (or not), so it will be the matrices in `Av` and `Bv`.

Same reasoning for the matrices `M` and `N`. They should have the same type of the `nep`. Indeed, if the linearization is explicitly computed, via `build_CORKPencil`, then if the `nep` is sparse, we want the linearization to be sparse. The linearization involve tensor products with the matrices `M` and `N`. Therefore they also should inherit the type from the `nep`.

Do you agree or you have a better argument to decoupled the matrix-types?

Reply:
I agree that we can have `eltype(Av)=eltype(Bv)`. 

I don't agree with this:
> Same reasoning for the matrices M and N. They should have the same type of the nep. Indeed, if the linearization is explicitly computed, via build_CORKPencil, then if the nep is sparse, we want the linearization to be sparse. The linearization involve tensor products with the matrices M and N. Therefore they also should inherit the type from the nep.

It might be okay to enforce `typeof(M)=typeof(N)` but not `typeof(M)=eltype(Av)`.   

Take the IAR-equivalent version. Then M and N will be extremely sparse matrices (independent if `Av` are sparse or not). If you couple `Av` and `N` you put limitations on the data type. 

That "the linearization is sparse" is not a good reasoning for a data type. The `CORKPencil` represents a  *factorization* of a linearization. The data type representing the factorization should be memory efficient.  If a *method* wants to construct the matrices corresponding to the `CORKPencil,` in a way that requires the types to be the same, the *method* should only be implemented for equal types, or (better) convert sparse to full matrices. Not all methods need to construct the actual pencil matrices.